packages/react-components/src/localization/localizationUtils.tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React from 'react';
+
+export interface IObjectMap<T> {
+  [key: string]: T;
+}
+
+/**
+ * Replace the pattern "{}" in str with the values passed in as vars
+ * eg: formatString("hello {name}. '{name}' is a rare name.", {name: "Foo"})
+ * return "hello Foo. 'Foo' is a rare name."
+ * @param {string} str The string to be formatted
+ * @param {Object} variables Variables to use to format the string
+ * @returns {string} formatted string
+ */
+export const formatString = (str: string, vars: IObjectMap<string>): string => {
+  if (!str) {
+    return '';
+  }
+  if (!vars) {
+    return str;
+  }
+
+  // regex to search for the pattern "{}"
+  const placeholdersRegex = /{(\w+)}/g;
+  return str.replace(placeholdersRegex, (_: string, k: string) => vars[k] || `{${k}}`);

```

Comment:
Where is this function being used? I'm worried that if a string has {name} in it already this will unintentionally replace chunks of the string it shouldn't

Reply:
I have removed this util function since it is not used.

Reply:
Same concern with `formatElements`, but given this is used sparingly, is an unlikely scenario to run into and the function itself is not publicly exposed (so we can make changes without them being breaking changes) this could be a fix for a later time if someone files a bug on it.